### PR TITLE
Sonar issues

### PR DESCRIPTION
**Pull request by lvermeulen**

### CODE REVIEW

File: src/StreetNameRegistry.Projections.LastChangedList/LastChangedProjections.cs [hunk 1]
```diff
@@ -196,9 +208,9 @@ protected override string BuildCacheKey(AcceptType acceptType, string identifier
             var shortenedAcceptType = acceptType.ToString().ToLowerInvariant();
             return acceptType switch
             {
-                AcceptType.Json => string.Format("legacy/streetname:{{0}}.{1}", identifier, shortenedAcceptType),
-                AcceptType.Xml => string.Format("legacy/streetname:{{0}}.{1}", identifier, shortenedAcceptType),
-                AcceptType.JsonLd => string.Format("oslo/streetname:{{0}}.{1}", identifier, shortenedAcceptType),
+                AcceptType.Json => $"legacy/streetname:{{{identifier}}}.{shortenedAcceptType}",
```
Comment (pgallik) [inline src/StreetNameRegistry.Projections.LastChangedList/LastChangedProjections.cs:211]: @ArneD didn't you revert these kinds of changes last time?

File: src/StreetNameRegistry.Projections.Legacy/StreetNameList/StreetNameListProjections.cs [hunk 2]
```diff
@@ -353,6 +354,9 @@ private static void UpdateHomonymAdditionByLanguage(StreetNameListItem entity, L
         private static void UpdateVersionTimestamp(StreetNameListItem streetNameListItem, Instant timestamp)
             => streetNameListItem.VersionTimestamp = timestamp;
 
-        private static void DoNothing() { }
+        private static async Task DoNothing()
```
Comment (pgallik) [inline src/StreetNameRegistry.Projections.Legacy/StreetNameList/StreetNameListProjections.cs:357]: Doesn't this do the same as the previous but make it more verbose?

File: src/StreetNameRegistry.Projections.LastChangedList/LastChangedProjections.cs (hunk 1 above)
Comment (ArneD) [inline src/StreetNameRegistry.Projections.LastChangedList/LastChangedProjections.cs:211]: This refactoring creates a different result.

File: src/StreetNameRegistry.Projections.Legacy/StreetNameName/StreetNameNameProjections.cs [hunk 3]
```diff
@@ -221,29 +222,28 @@ await context
...
+            When<Envelope<StreetNameStatusWasImportedFromCrab>>(async (context, message, ct) => await DoNothing());
         }
 
-        private static void UpdateNameByLanguage(StreetNameName streetNameName, Language? language, string name)
+        private static void UpdateNameByLanguage(StreetNameName streetNameName, Language? language, string? name)
         {
             switch (language)
             {
                 default:
-                case Language.Dutch:
```
Comment (ArneD) [inline src/StreetNameRegistry.Projections.Legacy/StreetNameName/StreetNameNameProjections.cs:246]: I prefer to let this stay: saying "I know dutch exists, it's the default behavior"
If someone new looks at the code and sees this, they might ask, should dutch be here?

File: src/StreetNameRegistry/StreetName/ValueObjects/StreetNameStatusLifetime.cs [hunk 4]
```diff
@@ -17,17 +17,65 @@ public StreetNameStatusLifetime(StreetNameStatus status, int? statusId, DateTime
             BeginDateTime = beginDateTime;
         }
 
+        public override bool Equals(object? obj)
+        {
```
Comment (pgallik) [inline src/StreetNameRegistry/StreetName/ValueObjects/StreetNameStatusLifetime.cs:21]: Can we not just use the Equals implementation from ValueObject itself?

@ArneD

File: src/StreetNameRegistry.Projections.Syndication/Municipality/MunicipalitySyndicationProjections.cs [hunk 5]
```diff
@@ -48,17 +48,18 @@ await context
...
+            {
                 return;
+            }
 
             foreach (var naam in gemeentenamen)
             {
                 switch (naam.Taal)
                 {
                     default:
-                    case Taal.NL:
```
Comment (ArneD) [inline src/StreetNameRegistry.Projections.Syndication/Municipality/MunicipalitySyndicationProjections.cs:61]: Same remark

**Review by ArneD — CHANGES_REQUESTED** [review]

File: src/StreetNameRegistry.Projections.LastChangedList/LastChangedProjections.cs (hunk 1 above)
Comment (ArneD) [reply src/StreetNameRegistry.Projections.LastChangedList/LastChangedProjections.cs:211]: No there was no refactoring done yet in streetname.

File: src/StreetNameRegistry/StreetName/ValueObjects/StreetNameStatusLifetime.cs (hunk 4 above)
Comment (ArneD) [reply src/StreetNameRegistry/StreetName/ValueObjects/StreetNameStatusLifetime.cs:21]: We do:
https://github.com/Informatievlaanderen/command-handling/blob/d5b2c7fbc92450e69f56cffb7880500b68fc495c/src/Be.Vlaanderen.Basisregisters.AggregateSource/ValueObject.cs#L164-L217

Comment (ArneD) [reply src/StreetNameRegistry/StreetName/ValueObjects/StreetNameStatusLifetime.cs:21]: just not all operators, mentioned here.

File: src/StreetNameRegistry.Projections.Legacy/StreetNameList/StreetNameListProjections.cs (hunk 2 above)
Comment (lvermeulen) [reply src/StreetNameRegistry.Projections.Legacy/StreetNameList/StreetNameListProjections.cs:357]: the issue is that DoNothing was an empty method AND was used in async lambda's

File: src/StreetNameRegistry.Projections.LastChangedList/LastChangedProjections.cs (hunk 1 above)
Comment (lvermeulen) [reply src/StreetNameRegistry.Projections.LastChangedList/LastChangedProjections.cs:211]: with the current code, identifier is not used -- with the changed code, identifier IS used

i can revert this, but it doesn't seem right. for AcceptType.Json, which result is expected?

File: src/StreetNameRegistry.Projections.Legacy/StreetNameName/StreetNameNameProjections.cs (hunk 3 above)
Comment (lvermeulen) [reply src/StreetNameRegistry.Projections.Legacy/StreetNameName/StreetNameNameProjections.cs:246]: it's implicit, no?

File: src/StreetNameRegistry/StreetName/ValueObjects/StreetNameStatusLifetime.cs (hunk 4 above)
Comment (lvermeulen) [reply src/StreetNameRegistry/StreetName/ValueObjects/StreetNameStatusLifetime.cs:21]: will fix

File: docs/StreetNameRegistry.Structurizr/Program.cs [hunk 6]
```diff
@@ -12,30 +12,30 @@ namespace StreetNameRegistry.Structurizr
...
     public static class CustomTags
     {
-        public static string Store = "Store";
-        public static string Event = "Event";
-        public static string Command = "Command";
-        public static string Https = "HTTPS";
-        public static string EntityFramework = "Entity Framework";
-        public static string SqlStreamStore = "SqlStreamStore";
-        public static string Direct = "Direct";
+        public static readonly string Store = "Store";
```
[the remark this answers is not in the document]
Comment (lvermeulen) [reply docs/StreetNameRegistry.Structurizr/Program.cs:29]: will fix

**Pull request merged**